compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -10129,6 +10133,11 @@ private STNode parseIntermediateClause(SyntaxKind nextTokenKind, boolean isRhsEx
             case JOIN_KEYWORD:
             case OUTER_KEYWORD:
                 return parseJoinClause(isRhsExpr);
+            case ORDER_KEYWORD:
+            case BY_KEYWORD:
+            case ASCENDING_KEYWORD:
+            case DESCENDING_KEYWORD:
+                return parseOrderByClause(isRhsExpr);
```

Comment:
Shall we add order-clause to documentation above; `intermediate-clause := from-clause | where-clause | let-clause`